query-planner-js/src/utilities/graphql.ts

```diff
@@ -97,21 +98,50 @@ export function astFromType(type: GraphQLType): TypeNode {
   }
 }
 
-export function printWithReducedWhitespace(ast: ASTNode): string {
-  return print(ast)
-    .replace(/\s+/g, ' ')
-    .trim();
+/**
+ * For lack of a "home of federation utilities", this function is copy/pasted
+ * verbatim across the federation, gateway, and query-planner packages. Any changes
+ * made here should be reflected in the other two locations as well.
+ *
+ * @param source A string representing a FieldSet
+ * @returns A parsed FieldSet
+ */
+export function parseSelections(source: string): ReadonlyArray<SelectionNode> {
+  const parsed = parse(`{${source}}`);
```

Comment:
Could be an opportunity to provide a more useful error message in the case of errors vai this `parseSelections` function, or possibly some specialized utility functions that produce valuable error messages in the event of `parse` errors; I'm just going to put this on the same radar by linking to this:

https://github.com/apollographql/federation/issues/679

e.g., perhaps a `parseKeys`, `parseRequires`, `parseProvides` might wrap this, or perhaps this can take a param.

Just a suggestion, not asking for it in this PR.